clang/include/clang/Basic/Builtins.td

```diff
@@ -833,6 +833,12 @@ def BuiltinVaStart : Builtin {
   let Prototype = "void(__builtin_va_list_ref, ...)";
 }
 
+def BuiltinC23VaStart : LangBuiltin<"C23_LANG"> {
```

Comment:
Is there any reason to restrict to C23?
The only thing that should be conditional is what `va_start` uses
In particular, C++ will align on C23 https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2025/p3348r2.pdf (page 9)

Reply:
This matches GCC's behavior and I don't think we want to extend support to earlier C modes given that the only use case is when the function signature is `(...)` which is invalid in earlier C modes.